user-data-store-server/src/main/java/com/wultra/security/userdatastore/service/DocumentService.java

```diff
@@ -97,7 +95,24 @@ public DocumentCreateResponse createDocument(final DocumentCreateRequest request
         updateDocumentHistory(documentEntity);
         audit("Created document for user ID: {}", userId);
 
-        return new DocumentCreateResponse(documentEntity.getId(), documentEntity.getDocumentDataId());
+        final DocumentEntity documentEntityFinal = documentEntity;
+
+        final List<EmbeddedPhotoCreateResponse> photosResponse = new ArrayList<>();
+        if (request.photos() != null && !request.photos().isEmpty()) {
+            request.photos().forEach(photoRequest -> {
+                final PhotoCreateResponse response = photoService.createPhoto(photoRequest, documentEntityFinal);
+                photosResponse.add(new EmbeddedPhotoCreateResponse(response.id()));
```

Comment:
🤔 I do not have a hard argument for that yet, but personally, I prefer `collect` over `forEach` with a side effect. One advantage is, that the collection is immutable. Something like this
```java
List<EmbeddedPhotoCreateResponse> photosResponse = request.photos().stream()
    .map(photoRequest -> photoService.createPhoto(photoRequest, documentEntityFinal))
    .map(response -> new EmbeddedPhotoCreateResponse(response.id()))
    .toList();
```